tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNet/HttpHeadersCollection.cs

```diff
@@ -25,31 +26,19 @@ public HttpHeadersCollection(IRequestHeaders headers)
             _headers = headers;
         }
 
-        public IEnumerable<string> GetValues(string name)
-        {
-            if (_headers.TryGetValues(name, out IEnumerable<string> values))
-            {
-                return values;
-            }
-
-            return Enumerable.Empty<string>();
-        }
+        public StringValues GetValues(string name) => _headers.TryGetValues(name, out var values) ?
+                                                          new StringValues(values.ToArray()) :
+                                                          StringValues.Empty;
```

Comment:
Isn't this actually _worse_ than before? 🤔 i.e. we're now allocating another array where we weren't before?

Reply:
Yeah, this is one place where this may be a regression. This is why I have a benchmarks TODO in the PR description 😄 

On the bright side, this is "legacy" ASP.NET, and it's replacing an enumerator allocation somewhere else for an array allocation here, so it's not _all_ bad.

Reply:
Also `values.ToArray()` returns the internal array without allocating if there is one. If the internal value is a single string, it returns `new[] { single_value }`.

Reply:
This is now gone with the new `StringEnumerable`.